src/framework/uploadImage/UploadImage.jsx

```diff
@@ -0,0 +1,114 @@
+
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import PreviewImage from './PreviewImage.jsx';
+
+class UploadImage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      file: null,
+      image: null,
+      hasErrors: false,
+      isloaded: false,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.hasLoaded = this.hasLoaded.bind(this);
+    this.onCloseImage = this.onCloseImage.bind(this);
+  }
+
+  componentWillMount() {
+    if (this.props.children) {
+      this.setState({
+        isloaded: true,
+      });
+    }
+  }
+
+  onCloseImage() {
+    // As default browser behaviour it doesnt allow you to reset what you selected
+    // by resetting the value you are able to reload the same image selected
+    document.getElementById('fileUploadInput').value = '';
```

Comment:
React refs? And when removing image from one with validation, the image actually stays so you cannot select the same image again.